### ISSUE TITLE
deno bundle <JavaScript file with triple-slash directive>

### ISSUE THREAD

**Issue by KSXGitHub** [denoland/deno#4539]
## Steps to reproduce

Run `deno bundle https://deno.land/x/once/index.js`

## Expected behavior

It prints a bundled JavaScript code

## Actual behavior

```
error TS5012: Cannot read file 'https://deno.land/x/once/index.js': Error: assert.

► 
error TS6053: File 'https://deno.land/x/once/index.js' not found.

► 

Found 2 errors.
```

---

I didn't try to bundle remote URL at first, instead, I tried bundling a local file (`deps.ts`) that import remote JS file. No compile errors occur, but a runtime error (cannot read property of `undefined`) was thrown when attempting to run it in the browser and in Node.js.

**Comment by kitsonk** [comment]
This doesn't have anything to do with it being remote.  The bug is related to trying to bundle a JavaScript file with a `/// <reference types="..." />` directive (and I suspect would also occur with ones where a `X-TypeScript-Types` header is supplied.  In these situations we replace the `.js` with the `.d.ts` file when we feed it to the compiler, but then the compiler can't access the `.js` file to generate the bundle.

**Comment by KSXGitHub** [comment]
@kitsonk I have changed the title of this issue because I trust what you said.

**Comment by quaos** [comment]
Hi,
Are there any updates on this?

**Comment by kitsonk** [comment]
No.

**Comment by matt-allan** [comment]
> I suspect would also occur with ones where a X-TypeScript-Types header is supplied

It does look like this happens when using the `X-TypeScript-Types` header. For example, using [skypack](https://docs.skypack.dev/code/deno):

```ts
import { h, render } from "https://cdn.skypack.dev/preact@^10.4.4?dts";

render(h('div', null, 'hello world'), document.body);
```

```json
// tsconfig.json
{
    "compilerOptions": {
      "jsx": "react",
      "jsxFactory": "h",
      "lib": [
          "DOM",
          "DOM.Iterable",
          "ESNext"
      ]
    }
  }
```

```bash
deno bundle -c tsconfig.json ./app.ts ./app.bundle.js
```

The [bundle](https://gist.github.com/matt-allan/bcaba41630d2c640a868fe0c7eb4d73b) is missing Preact, but works fine if you remove the `?dts` query param.

**Comment by kitsonk** [comment]
This was fixed in Deno 1.5
